packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -498,20 +503,70 @@ export default class Meeting extends StatelessSparkPlugin {
     this.setCorrelationId(this.id);
     // this.state.join();
     this.spark.meetings.notifyRinging(this, OUTGOING);
-    return MeetingUtil.joinMeeting(this)
+    return MeetingUtil.joinMeeting(this, options)
       .then((response) => {
         this.setLocus(response);
         this.spark.meetings.notifyRinging(this, null, false);
         // this.state.connect();
-        // TODO: different scenarios for meeting , claimed and unclaimed
       })
       .catch((err) => {
         this.spark.meetings.notifyRinging(this, null, false);
+        if (MeetingUtil.isPinOrGuest(err)) {
+          if (MeetingUtil.hasOwner(this.meetingInfo)) {
+            return MeetingUtil.joinMeeting(this, {
+              moderator: MeetingUtil.isOwnerSelf(this.owner, this.userId) ? MODERATOR_TRUE : MODERATOR_FALSE
+            })
+              .then((response) => {
+                this.setLocus(response);
+                // this.state.connect();
+              });
+          }
+          // request host pin or non host for unclaimed PMR
+          this.trigger(EVENT_TRIGGERS.LOBBY_HOST_GUEST, {
+            error: err,
+            message: 'The meeting is locked. This is expected behavior. Call #joinAsHost or #joinAsGuest',
+            joinAsHostOrGuest: true
+          });
+          // request host pin or non host for unclaimed PMR
+          // TODO: deal with Error object
+          // eslint-disable-next-line
+          return Promise.reject({
+            error: err,
+            message: 'The meeting is locked. This is expected behavior. Call #joinAsHost or #joinAsGuest',
+            joinAsHostOrGuest: true
+          });
+        }
         this.logger.log('Meeting:index#join --> Error joining the call on getLocalMedia, ', err);
-        throw err;
+        return Promise.reject(new Error({
+          error: err,
+          message: 'Meeting:index#join --> Error joining the call on getLocalMedia'
+        }));
       });
   }
 
+  /**
+   * @param {Number} hostPin a 6 digit webex hostpin for unclaimed PMR host pin
+   * @param {Boolean} moderator optional
+   * @returns {Promise}
+   * @throws {Error} if hostPin is not passed
+   * @memberof Meeting
+   */
+  joinLobbyAsHost(hostPin, moderator = MODERATOR_TRUE) {
```

Comment:
thoughts on naming scheme here?

Reply:
Is it joining the lobby as host or joining the meeting as host?

Reply:
Do we need the second flag? What happens if I call `joinLobbyAsHost(34356, false)`?

Reply:
Can we use `join` instead here with different args?